config/config.go

```diff
@@ -19,6 +20,51 @@ type Config struct {
 	HTTPListener      string        `long:"http-listener" description:"host:port to listen for HTTP connections"`
 	BitcoinDisableTLS bool          `long:"bitcoin-disable-tls" description:"disable TLS for RPC connections"`
 	PollInterval      time.Duration `long:"retry-interval" description:"interval in seconds to recheck Babylon finality of block"`
+	BatchSize         uint64        `long:"batch-size" description:"number of blocks to process in a batch"`
+}
+
+func (c *Config) Validate() error {
+	// Required fields
+	if c.L2RPCHost == "" {
+		return fmt.Errorf("l2-rpc-host is required")
+	}
+	if c.BitcoinRPCHost == "" {
+		return fmt.Errorf("bitcoin-rpc-host is required")
+	}
+	if c.BitcoinRPCUser == "" {
+		return fmt.Errorf("bitcoin-rpc-user is required")
+	}
+	if c.BitcoinRPCPass == "" {
+		return fmt.Errorf("bitcoin-rpc-pass is required")
+	}
```

Comment:
iirc we used the Ankr RPC that doesn't need user and password in our e2e test and it's working